### ISSUE TITLE
Regression: -f '*/file' should match 'file'

### ISSUE THREAD

**Issue by ngirard** [watchexec/watchexec#258]
With watchexec 1.15.1, this used to work:

```sh
watchexec -p -f '*/justfile' -f '*/.env' "do domething"
```

With 1.18.6, it requires rewriting to:

```sh
watchexec -p -f '/justfile' -f '/.env' "do domething"
```

**Comment by passcod** [comment]
No, see, if it's not an intended change, it's called a bug.

**Comment by ngirard** [comment]
Awww... sorry about that...

**Comment by passcod** [comment]
That's alright. The goal of 1.18 is to get all this stuff fixed. If it's not in the big list in the 1.18.0 release, it's probably a bug.

**Comment by ngirard** [comment]
Well hold on! I consider watchexec and its lib as important for the Rust ecosystem as the ripgrep-related crates, and all your efforts are greatly appreciated !

**Comment by passcod** [comment]
Can you check a verbose log and see where the "project origin" is being discovered as? Is it the working directory or somewhere above? It shouldn't matter anyway, and indeed I would expect the `/foo` invocation _not_ to work in this case, but just to check...

**Comment by ngirard** [comment]
> Specifically, when the current/working directory isn't in version control

Actually I inferred this from the release notes to v1.18.6 ; but performing a `git init` doesn't make any difference using both v0.18.5 and 0.18.6. Sorry about that.

**Comment by passcod** [comment]
Alright, this is the test I'll be working to:

```rust
#[tokio::test]
async fn leading_single_glob_file() {
	let filterer = filt(&["*/justfile"], &[], &[]).await;

	filterer.file_does_pass("justfile");
	filterer.file_does_pass("foo/justfile");
	filterer.file_doesnt_pass("notfile");
	filterer.file_doesnt_pass("not/thisfile");
}
```

Also aiming for .7

**Comment by passcod** [comment]
After review... unfortunately, `*/file` should _not_ match `file`. That is, within a directory structure like:

```
 .
├── foo
└── lib
   └── foo
```

The filter pattern `*/foo` will match `lib/foo`, but not `foo`. This is consistent with git ignore behaviour.

**However**, I recognise this _is_ a breaking change. So, just like some other filtering behaviours which are arguably 1.x bugs but would break compat, I will add an exception to the engine that specifically allows this, _for the 1.x branch_. When 2.x comes out, this will be one of the breaking changes.

**Comment by passcod** [comment]
Apologies for the delay! This is now released [in 1.18.7](https://github.com/watchexec/watchexec/releases/tag/cli-v1.18.7).

**Comment by ngirard** [comment]
No worries, Félix. Thanks for your work !
